test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/WipeTokenAccountPrecompileSuite.java

```diff
@@ -0,0 +1,160 @@
+import static com.hederahashgraph.api.proto.java.TokenType.FUNGIBLE_COMMON;
+import static com.hederahashgraph.api.proto.java.TokenType.NON_FUNGIBLE_UNIQUE;
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.TokenID;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class WipeTokenAccountPrecompileSuite extends HapiApiSuite {
+    private static final Logger log = LogManager.getLogger(WipeTokenAccountPrecompileSuite.class);
+    private static final String WIPE_CONTRACT = "WipeTokenAccount";
+    private static final String ACCOUNT = "anybody";
+    private static final String WIPE_KEY = "wipeKey";
+    private static final String MULTI_KEY = "purpose";
+    private static final long GAS_TO_OFFER = 4_000_000L;
+
+    public static void main(String... args) {
+        new WipeTokenAccountPrecompileSuite().runSuiteSync();
+    }
+
+    @Override
+    protected Logger getResultsLogger() {
+        return log;
+    }
+
+    @Override
+    public List<HapiApiSpec> getSpecsInSuite() {
+        return allOf(List.of(wipeFungibleTokenHappyPath(), wipeNonFungibleTokenHappyPath()));
+    }
+
+    private HapiApiSpec wipeFungibleTokenHappyPath() {
+        final AtomicReference<AccountID> accountID = new AtomicReference<>();
+        final AtomicReference<TokenID> vanillaTokenID = new AtomicReference<>();
+
+        return defaultHapiSpec("WipeFungibleTokenHappyPath")
+                .given(
+                        newKeyNamed(WIPE_KEY),
+                        cryptoCreate(ACCOUNT).exposingCreatedIdTo(accountID::set),
+                        cryptoCreate(TOKEN_TREASURY),
+                        tokenCreate(VANILLA_TOKEN)
+                                .tokenType(FUNGIBLE_COMMON)
+                                .treasury(TOKEN_TREASURY)
+                                .wipeKey(WIPE_KEY)
+                                .initialSupply(1_000)
+                                .exposingCreatedIdTo(id -> vanillaTokenID.set(asToken(id))),
+                        uploadInitCode(WIPE_CONTRACT),
+                        contractCreate(WIPE_CONTRACT),
+                        tokenAssociate(ACCOUNT, VANILLA_TOKEN),
+                        cryptoTransfer(moving(500, VANILLA_TOKEN).between(TOKEN_TREASURY, ACCOUNT)))
+                .when(
+                        withOpContext(
+                                (spec, opLog) ->
+                                        allRunFor(
+                                                spec,
+                                                contractCreate(WIPE_CONTRACT)
+                                                        .bytecode(WIPE_CONTRACT),
```

Comment:
It seems you create `WIPE_CONTRACT` twice. Also `bytecode` is not needed anymore to be passed explicitly.